python/ray/train/v2/api/config.py

```diff
@@ -123,9 +123,11 @@ class RunConfig(RunConfigV1):
             prefix stripped (e.g., `s3://bucket/path` -> `bucket/path`).
         failure_config: Failure mode configuration.
         checkpoint_config: Checkpointing configuration.
+        callbacks: [DeveloperAPI] A list of callbacks that the Ray Train controller
+            will invoke during training.
     """
 
-    callbacks: Optional[List["Callback"]] = None
+    callbacks: Optional[List["RayTrainCallback"]] = None
```

Comment:
Should we keep this as `UserCallback` for now to prevent misuse, and expand as needed?

Reply:
It's a bit confusing if we have a `UserCallback` class but it's not exposed in the public API.

Reply:
What about making the type `UserCallback` but then keeping this https://github.com/ray-project/ray/pull/49819/files#diff-730d6551cfa9d3b55d3d413ce0500fad427b6409bc47d7e103f4f586341c3b2eR210-R214

So that workarounds / custom callbacks are still possible for workarounds and in unit tests. It will just break the typechecking on the editor.

Reply:
Yeah that works for now.